web-ui/controller/git_forge.ml

```diff
@@ -361,5 +362,5 @@ module Make_API (Api : Api) = struct
     Client.Commit.status commit_cap >>!= fun status ->
     Client.Commit.jobs commit_cap >>!= fun jobs ->
     let build_status = Client.State.from_build_status status in
-    Api.list_steps ~build_status ~jobs
+    Api.list_steps ~build_status ~jobs ~step_route_prefix
```

Comment:
I am surprised you don't derive this from the View.Api module used.

Reply:
Etienne brought this up as well - pls see [this thread](https://github.com/ocurrent/ocaml-ci/pull/576#pullrequestreview-1165509975) for my rationale for keeping this in the router.

Reply:
@tmcgilchrist I've worked the ideas that came up following our IRL conversation regarding changes to the current approach to functorising `View.Api` (to also include concerns like the prefix and possibly the `step_route_prefix`). 

I've decided to merge this as it is ... but to put up a separate PR that will try out this approach.